packages/api-derive/src/chain/util.ts

```diff
@@ -38,9 +38,7 @@ export function getAuthorDetails (header: Header, queryAt: QueryableStorage<'rxj
     (log.isPreRuntime && log.asPreRuntime[0].isNimbus && log.asPreRuntime[1])
   )) || null;
 
-  const validators = (queryAt.session)
-    ? queryAt.session.validators()
-    : null;
+  const validators = (queryAt.session) ? queryAt.session.validators() : null;
```

Comment:
This is where the issue starts and why we need a cast below -

The types between these two are different (the one being an Observable, the other not), so this pushed through to the actual returntype. Originally we have `Observable<AccountId> | Observable<null>`, now it is `Observable<AccountId> | null`

Reply:
CI looks to have succeeded with the last run.
Does this work or do you prefer the cast-less version?

Reply:
the version before my last commit didn't actually work (despite successful lint and build).

for whatever reason, having a single return statement at the end of the method and populating the final element of the combinelatest array arg with either a `mappedAuthor` (nimbus) or `of(null)` was refusing to lint and build.

my efforts to replace `combineLatest` were flawed and as it turns out, also unnecessary.

having three separate return paths (nimbus/moonbeam, nimbus/manta, generic) does both lint and build and also does not throw deprecation errors on `combineLatest`. i can't say i understand why, but i assume that only specific invocation signatures are deprecated.

earlier versions (of my changes in this pr) were also flawed in the type declaration of the return value for `session.queuedKeys()`, which in the case of manta/nimbus is as simple as `Array<[AccountId, { nimbus: Address }]>`. something i only learned through a bunch of trial and error here.